src/effects/backends/builtin/instrumentalseparationeffect.cpp

```diff
@@ -0,0 +1,243 @@
+            engineParameters.sampleRate(), kVocalNotch2Freq, kNotchQ);
+}
+
+InstrumentalSeparationGroupState::~InstrumentalSeparationGroupState() {
+    delete m_pBassEnhancer;
+    delete m_pMidCut;
+    delete m_pHighEnhancer;
+    delete m_pVocalNotch1;
+    delete m_pVocalNotch2;
+}
+
+void InstrumentalSeparationEffect::loadEngineEffectParameters(
+        const QMap<QString, EngineEffectParameterPointer>& parameters) {
+    m_pIntensity = parameters.value("intensity");
+    m_pBassBoost = parameters.value("bassboost");
+    m_pHighBoost = parameters.value("highboost");
+}
+
+void InstrumentalSeparationGroupState::setFilters(
+        mixxx::audio::SampleRate sampleRate,
+        double intensity,
+        double bassBoost,
+        double highBoost) {
+    // Update filters only when parameters change significantly
+    if (std::abs(intensity - m_oldIntensity) > 0.01 ||
+        std::abs(bassBoost - m_oldBassBoost) > 0.05 ||
+        std::abs(highBoost - m_oldHighBoost) > 0.05) {
+        
+        // Boost bass frequencies
+        double bassGain = bassBoost * 12.0; // Up to +12dB
+        m_pBassEnhancer->setFrequencyCorners(
+                sampleRate, kBassFreq, kBoostQ, bassGain);
+        
+        // Cut mid frequencies (vocal range) based on intensity
+        double midCut = -(intensity * 15.0); // Up to -15dB
+        m_pMidCut->setFrequencyCorners(
+                sampleRate, kMidFreq, kBoostQ, midCut);
+        
+        // Boost high frequencies
+        double highGain = highBoost * 10.0; // Up to +10dB
+        m_pHighEnhancer->setFrequencyCorners(
+                sampleRate, kHighFreq, kBoostQ, highGain);
+        
+        // Set vocal notch filters (inverse gain for notch)
+        m_pVocalNotch1->setFrequencyCorners(
+                sampleRate, kVocalNotch1Freq, kNotchQ);
+        m_pVocalNotch2->setFrequencyCorners(
+                sampleRate, kVocalNotch2Freq, kNotchQ);
+        
+        m_oldIntensity = intensity;
+        m_oldBassBoost = bassBoost;
+        m_oldHighBoost = highBoost;
+    }
+}
+
+void InstrumentalSeparationEffect::processChannel(
+        InstrumentalSeparationGroupState* pState,
+        const CSAMPLE* pInput,
+        CSAMPLE* pOutput,
+        const mixxx::EngineParameters& engineParameters,
+        const EffectEnableState enableState,
+        const GroupFeatureState& groupFeatures) {
+    Q_UNUSED(groupFeatures);
+
+    const CSAMPLE intensity = m_pIntensity ? m_pIntensity->value() : 0.0;
+    const CSAMPLE bassBoost = m_pBassBoost ? m_pBassBoost->value() : 0.5;
+    const CSAMPLE highBoost = m_pHighBoost ? m_pHighBoost->value() : 0.5;
+
+    if (enableState == EffectEnableState::Disabling) {
+        // Ramp to dry signal
+        SampleUtil::copy(pOutput, pInput, engineParameters.samplesPerBuffer());
+        return;
+    }
+
+    if (intensity <= 0.0) {
+        // Pass through when no intensity
+        SampleUtil::copy(pOutput, pInput, engineParameters.samplesPerBuffer());
```

Comment:
The same here. We need to run the last callback and than ramp. 

The issue is that these filter introduce a group delay. A small signal shift. This causes the dry and the whet signal do not fit together back to back. Ramping is required to lind the click of the resulting rectangular jump.